blues_refactor/ncmc.py

```diff
@@ -0,0 +1,216 @@
+        positions = copy.deepcopy(initial_positions)
+
+        # Define random rotational move on the ligand
+        rand_quat = mdtraj.utils.uniform_quaternion()
+        rand_rotation_matrix = mdtraj.utils.rotation_matrix_from_quaternion(rand_quat)
+
+        #multiply lig coordinates by rot matrix and add back COM translation from origin
+        rot_move =  np.dot(reduced_pos, rand_rotation_matrix) * unit.nanometers + com
+
+        # Update ligand positions in nc_sim
+        for index, atomidx in enumerate(atom_indices):
+            positions[atomidx] = rot_move[index]
+        self.nc_sim.context.setPositions(positions)
+
+        return self.nc_sim
+
+class Simulation(object):
+    def __init__(self, sims, model, **opt):
+        self.md_sim = sims.md
+        self.alch_sim = sims.alch
+        self.nc_context = sims.nc.context
+        self.nc_integrator = sims.nc.context._integrator
+        self.model = model
+        self.accept = 0
+        self.reject = 0
+        self.accept_ratio = 0
+
+        self.current_stepNC = 0
+        self.current_iter = 0
+        self.current_stepMD = 0
+
+        self.current_state = {}
+
+        self.temperature = opt['temperature']
+        kB = unit.BOLTZMANN_CONSTANT_kB * unit.AVOGADRO_CONSTANT_NA
+        kT = kB * opt['temperature']
+        beta = 1.0 / kT
+        self.beta = beta
+
+        self.work_keys = ['total_work', 'lambda', 'shadow_work',
+                          'protocol_work', 'Eold', 'Enew','Epert']
+
+        self.state_keys = { 'getPositions' : True,
+                       'getVelocities' : True,
+                       'getForces' : False,
+                       'getEnergy' : True,
+                       'getParameters': True,
+                       'enforcePeriodicBox' : True}
+
+
+    def setCurrentState(self, simkey, stateinfo):
+        current_state = {}
+        for k,v in stateinfo.items():
+            current_state[k] = v
+        self.current_state[simkey] = current_state
+
+    def getStateInfo(self, context, parameters):
+        stateinfo = {}
+        state  = context.getState(**parameters)
+        stateinfo['iter'] = int(self.current_iter)
+        stateinfo['positions'] =  state.getPositions(asNumpy=True)
+        stateinfo['velocities'] = state.getVelocities(asNumpy=True)
+        stateinfo['potential_energy'] = state.getPotentialEnergy()
+        stateinfo['kinetic_energy'] = state.getKineticEnergy()
+        return stateinfo
+
+    def getWorkInfo(self, nc_integrator, parameters):
+        workinfo = {}
+        for param in parameters:
+            workinfo['param'] = nc_integrator.getGlobalVariableByName(param)
+        return workinfo
+
+    def chooseMove(self, md_state0, nc_state0, nc_state1):
+        log_ncmc = self.nc_integrator.getLogAcceptanceProbability(self.nc_context)
+        randnum =  math.log(np.random.random())
+
+        ### Compute Alchemical Correction Term
+        if not np.isnan(log_ncmc):
```

Comment:
This originally had the `if alchemical_correction==True` statement with checking if `log_ncmc` is NaN. So I _think_ that if you were to add the alchemical correction term, it won't let you add it to NaN. Since I've removed the alchemical_correction switch, should we turn NaN into 0 so that we can still add the alchemical correction term?